File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    answer = len(citations)
    
    citations.sort(reverse=True)
    
    while answer >= 0:
        if (answer < len(citations) and citations[answer-1] >= answer and citations[answer] <= answer) or (answer == len(citations) and citations[answer-1] >= answer):
            break 
        answer -= 1
        
    return answer

File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        int answer = citations.length;
        int len = citations.length;
        
        Arrays.sort(citations);
        
        while(answer > 0){
            if((answer == len && citations[len-answer] >= answer) 
                || (answer < len && citations[len-answer] >= answer && citations[len-answer-1] <= answer)) break;
            answer--;
        }
        return answer;
    }
}